src/modules/git_status.rs

```diff
@@ -570,23 +570,38 @@ impl RepoStatus {
     }
 
     fn set_ahead_behind_for_each_ref(&mut self, mut s: &str) {
-        s = s.trim_matches(|c| c == '[' || c == ']');
-
-        for pair in s.split(',') {
-            let mut tokens = pair.trim().splitn(2, ' ');
-            if let (Some(name), Some(number)) = (tokens.next(), tokens.next()) {
-                let storage = match name {
-                    "ahead" => &mut self.ahead,
-                    "behind" => &mut self.behind,
-                    _ => return,
-                };
-                *storage = number.parse().ok();
+        let re = Regex::new(r"^([^ ]+) (\[(.*)\])?$").unwrap();
```

Comment:
This regex should not be recreated for every invocation, please try to avoid regexes or if that is not possible cache the regex.

Reply:
Done. And funnily enough, the diff is even shorter now 😁 

Let me know if there's anything else to do